File: ExamUserInterface/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import base_url from './helper';
@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http:HttpClient) { }
  
public getQuiz(){
  return this.http.get(`${base_url}/quiz/`);
}
public addQuiz(quiz) {
  return this.http.post(`${base_url}/quiz/create`, quiz);
}

public deleteQuiz(qId) {
  return this.http.delete(`${base_url}/quiz/${qId}`);
}

public getQuizById(qId) {
  return this.http.get(`${base_url}/quiz/${qId}`);
}

public updateQuiz(quiz) {
  return this.http.put(`${base_url}/quiz/`, quiz);
}

public getQuizzesByCategory(cid){
  return this.http.get(`${base_url}/quiz/category/${cid}`);
}
public getActiveQuizzes(){
  return this.http.get(`${base_url}/quiz/active`);
}
public getQuizzesByCategoryAndActive(cid){
  return this.http.get(`${base_url}/quiz/active/category/${cid}`);
}
}

 

File: ExamServer/src/main/java/com/exam/service/QuizService.java
package com.exam.service;

import java.util.Set;

import com.exam.entity.Category;
import com.exam.entity.Quiz;

public interface QuizService {

	public Quiz addQuiz(Quiz quiz);
	public Quiz updateQuiz(Quiz quiz);
	public Set<Quiz> getQuizzes();
	public Quiz getQuiz (long quizId);
	public void deleteQuiz(long quizId);
	public Set<Quiz> getQuizByCategory(Category category);
	public Set<Quiz> getActiveQuiz();
	public Set<Quiz> getQuizByCategoryandActive(Category category);
}
